### ISSUE TITLE
Wireless controllers on Android TV not usable. (NVIDIA Shield).

### ISSUE THREAD

**Issue by michael-conrad** [godotengine/godot#45712]
* Godot version: v3.2.4.rc1.official
* OS/device including version: Android/NVIDIA Shield TV (2015)/Firmware 8.2.2
* Issue description: Wireless Joypads do not work correctly
* Steps to reproduce: Install Joypad Demo with controller mapping on NVIDIA Shield TV. Use Joypad.
* Minimal reproduction project: Joypad Demo project.

There is something very much wrong with the wireless controllers on Android TV. Tested with OUYA and NVIDIA wireless controllers.

I do have a mapping for the NVIDIA controller being applied.

Both the OUYA and NVIDIA controllers, at rest, will show both the leftaxisx and rightaxisx pulled left. Other weirdness as well.

I have an X-Box 360 USB controller, and it works fine.

This pretty much kills Godot for Android TV. At least for the NVIDIA Shield.

I downloaded the Joypad demo from the assets store and added the mapping for the NVIDIA controller.

Here are screenshots of Joypad inputs showing the bad axis values while the joypads are not being used.

![image](https://user-images.githubusercontent.com/5229492/106913394-f2a48b80-66d1-11eb-8d43-f55df4435a76.png)

![image](https://user-images.githubusercontent.com/5229492/106913440-fd5f2080-66d1-11eb-9f80-57d8ad85653f.png)

The code changes to Joypad Demo:

```gdscript
func _ready():
	set_physics_process(true)
	Input.connect("joy_connection_changed", self, "_on_joy_connection_changed")
	Input.add_joy_mapping("4e564944494120436f72706f72617469,NVIDIA Xbox Style Controller,a:b0,b:b1,back:b15,dpdown:h0.4,dpleft:h0.8,dpright:h0.2,dpup:h0.1,guide:b5,leftshoulder:b9,leftstick:b7,lefttrigger:a4,leftx:a0,lefty:a1,rightshoulder:b10,rightstick:b8,righttrigger:a5,rightx:a2,righty:a3,start:b6,x:b2,y:b3,platform:Android,", true)
```

I also changed the behavior so that the active device being displayed is auto switched to any device added. (Saves from having to use keyboard/mouse on TV).

```gdscript
func _on_joy_connection_changed(device_id, connected):
	if device_id == cur_joy:
		joypad_name.set_text(Input.get_joy_name(device_id) if connected else "")
	if connected:
		joypad_number.value=device_id
```

_Originally posted by @michael-conrad in https://github.com/godotengine/godot/issues/12696#issuecomment-773389083_

**Comment by Calinou** [comment]
@michael-conrad Please fill in the [issue template](https://github.com/godotengine/godot/blob/master/.github/ISSUE_TEMPLATE/bug_report.md) when opening issues instead of replacing it.

**Comment by michael-conrad** [comment]
Sorry, it didn't display, I used the comment->new ticket menu option. I'll edit the initial submission now.

**Comment by michael-conrad** [comment]
I've tracked down the issue to the Java shim on Android.

Some of the axis values show up more than once, but with different source values.

I've also run into an issue with the OUYA where the main game engine won't accept more than 10 axis values. The OUYA shows up as a 12 axis device, reporting it's dpad as axis and not hat.

**Comment by akien-mga** [comment]
Fixed by #45771.
